Source/Scene/Model.js

```diff
@@ -4369,12 +4389,13 @@ define([
 
         releaseCachedGltf(this);
 
-        // Check if ClippingPlaneCollection was already destroyed, for instance, by Cesium3DTileset
-        var clippingPlaneCollection = this.clippingPlanes;
-        if (defined(clippingPlaneCollection) && !clippingPlaneCollection.isDestroyed()) {
-            // If not, only destroy if the ClippingPlaneCollection is not owned by a Cesium3DTileset
-            clippingPlaneCollection.checkDestroy(this);
+        // Only destroy the ClippingPlaneCollection if this is the owner - if this model is part of a Cesium3DTileset,
+        // _clippingPlanes references a ClippingPlaneCollection that this model does not own.
+        var clippingPlaneCollection = this._clippingPlanes;
+        if (defined(clippingPlaneCollection) && !clippingPlaneCollection.isDestroyed() && clippingPlaneCollection._owner === this) {
```

Comment:
Perhaps add a getter for `ClippingPlaneCollection.owner` so it doesn't have to access a private var.